pymeasure/instruments/common_base.py

```diff
@@ -279,7 +279,7 @@ def wait_for(self, query_delay=0):
         """
         raise NotImplementedError("Implement in subclass!")
 
-    def ask(self, command, query_delay=0):
+    def ask(self, command, query_delay=0, **kwargs):
```

Comment:
Yes. A mock call might suffice, or writing our own ask method, which stores the kwargs handed to it.

Reply:
I have added a test which tests if the kwarg is passed through to `read`. Please have a look if this is ok for you. It of course requires `ask` to stay a `write`/`read` sequence.

Reply:
Oh, I just realised, that it is in the pymeasure core.

Why should the kwargs go to the read and not to write (or both)?
Because there is no simple yet general solution for that question, I did not add kwargs. The most practical solution is, that an instrument (like yours, I guess), which requires more arguments in ask, overrides ask. That way you can decide, which arguments go where.

Reply:
ok, I understand your concern and I will move the `ask` implementation to the Instrument. 
But would you at least agree that values shall forward the `kwargs` to ask in `CommonBase`? 
Otherwise one would also need to reimplement `values`.

Reply:
As there is no other good recipient of kwargs in values than "ask", I'm fine with handing kwargs to ask.

Reply:
It is good, how values handles the kwargs, but `ask` should not accept kwargs and drop them silently.

Please remove the **kwargs from the `ask` method.

That way a user sees, that his arguments are not processed (because ask raises an error).